src/utils/maps.ts

```diff
@@ -18,40 +20,45 @@ export function mapEdgesToItems<T>(data?: EdgesType<T>): T[] {
   return data.edges.map(({ node }) => node);
 }
 
-export function mapCountriesToChoices(
-  countries: ShopInfo_shop_countries[]
-): Array<SingleAutocompleteChoiceType | MultiAutocompleteChoiceType> {
+export function mapCountriesToChoices(countries: ShopInfo_shop_countries[]) {
   return countries.map(country => ({
     label: country.country,
     value: country.code
   }));
 }
 
-export function mapPagesToChoices(
-  pages: SearchPages_search_edges_node[]
-): Array<SingleAutocompleteChoiceType | MultiAutocompleteChoiceType> {
+export function mapPagesToChoices(pages: SearchPages_search_edges_node[]) {
   return pages.map(page => ({
     label: page.title,
     value: page.id
   }));
 }
 
-export function mapNodeToChoice(
-  nodes: Array<Node & Record<"name", string>>
-): Array<SingleAutocompleteChoiceType | MultiAutocompleteChoiceType> {
+type ExtendedNode = Node & Record<"name", string>;
+export function mapNodeToChoice<T extends ExtendedNode>(
+  nodes: T[]
+): Array<SingleAutocompleteChoiceType<string>>;
+export function mapNodeToChoice<T extends ExtendedNode, K extends ChoiceValue>(
+  nodes: T[],
+  getterFn: (node: T) => K
+): Array<SingleAutocompleteChoiceType<K>>;
+export function mapNodeToChoice<T extends ExtendedNode>(
+  nodes: T[],
+  getterFn?: (node: T) => any
+) {
   if (!nodes) {
     return [];
   }
 
   return nodes.map(node => ({
     label: node.name,
-    value: node.id
+    value: getterFn ? getterFn(node) : node.id
   }));
 }
 
 export function mapSlugNodeToChoice(
```

Comment:
I think since this can be now well handled by mapNodeToChoice which is used in this pr exactly for slug item, in order to keep it consistent we should either: 
- also remove `mapSlugNodeToChoice` and use mapNodeToChoice instead
- make mapSlugNodeToChoice use mapNodeToChoice underneath `(nodes) => mapNodeToChoice(nodes, node => node.slug)` and use that instead of manually doing it each time

Imo since the slug nodes are second most commonly mapped to choices in the app (right after items that use `id`) it'd be nice to keep it separate, not to rewrite with each usage `mapNodeToChoice(availableChannels, channel => channel.slug)`. And I'd keep manually adding the getter function when items we're mapping are singularly used. Wdyt?

Reply:
Second options looks good, I wasn't aware there was a dedicated fn for that. Nice catch